libbpf-tools/utils.h

```diff
@@ -0,0 +1,14 @@
+#ifndef __UTILS_H
+
+#define INVALID_UID ((uid_t)-1)
+#define INVALID_GID ((gid_t)-1)
+
+static __always_inline bool valid_uid(uid_t uid) {
+	return uid != INVALID_UID;
+}
+
+static __always_inline bool valid_gid(gid_t gid) {
```

Comment:
nothing is using valid_gid() and INVALID_GID, please drop it. No need for utils.h as well, just put valid_uid inside opensnoop.bpf.c

Reply:
I added them because I think it's logical if we have `uid` filter, then `gid` filter is something that has to be. At least most *nix tools has that convention.

But I agree, it would be better to add them when there will be tools that use it. I'll remove them.

Regarding that there is no need in `utils.h`, I recalled why I thought it's bad idea to use only one variable for uid filtering, it's primarily duplication. Basically, in this case we need [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming))  that will repeat both in `opensnoop.c` (we need to initialize variable with invalid uid if we are not doing uid filtering) and `opensnoop.bpf.c`. 

Moreover, porting other tools (e.g. `execsnoop`, `bindsnoop`) would require same functions, since they also implementing uid filtering and it's better to share this code between them. Also, in some sense, python tools has something [similar](https://github.com/iovisor/bcc/blob/master/src/python/bcc/utils.py#L61).

But, maybe it's something that should be added in separate PR.

Reply:
Yeah, let's postpone all this until we really need it. Duplicating INVALID_UID definition is not a big deal, IMO.